### DISCUSSION TITLE
calculating Yearly returns

### DISCUSSION THREAD

**Discussion by BaijuThomasAV** [atoti/atoti#860] [General]
Im trying to build a 1 Year returns  with a data set that has 3 columns - Dates / Cities /and Prices , i have prices for 5 Years across 5 cities, so for a 1Y return per city, i need to do the below but i get a AttributeError: 'Measure' object has no attribute 'lag' . not sure how to fix this .Thanks.

#  return for 1 year for each city
return_1y = cube.query(
    m["Prices.SUM"],  
    levels=[l["Cities"]],
).sort_values("Prices.SUM", ascending=False)

# Calculate the 1Y return based on the latest price and the price from 12 months ago
latest_price = cube.query(m["Prices.SUM"], levels=[l["Cities"], l["Dates"]]).tail(1)
price_12_months_ago = cube.query(m["Prices.SUM"].lag(amount=12), levels=[l["Cities"], l["Dates"]]).tail(1)

**Comment by OPeyrusse** [comment]
Hello @BaijuThomasAV ,

can you provide a sample notebook, showing what your data is and how your cube is setup?
With only the error, it would be a large investigation to find what went wrong.

**Comment by BaijuThomasAV** [comment]
Thanks. i managed to solve it . basically, below was my raw data and i wanted to find out what the 1year , 5 year, 10Year and 20Y return was. below calculations helped me get the answer. 

Raw Data -sample
Dates | Cities | Prices
-- | -- | --
1/1/1995 | City of London | 91449
2/1/1995 | City of London | 82203
3/1/1995 | City of London | 79121
4/1/1995 | City of London | 77101
5/1/1995 | City of London | 84409
6/1/1995 | City of London | 94901
7/1/1995 | City of London | 110128
8/1/1995 | City of London | 112329
9/1/1995 | City of London | 104473
10/1/1995 | City of London | 108038
11/1/1995 | City of London | 117636


Calculation
l["Dates"].order = tt.NaturalOrder(ascending=False)

`m["1YRet"] = ((m["PricesPerCity"]/(tt.shift(m["PricesPerCity"], h["Dates"], offset=12))) - 1 )* 100

m["5YRet"] = ((m["PricesPerCity"]/(tt.shift(m["PricesPerCity"], h["Dates"], offset=60)) - 1 ))* 100

m["10YRet"] = ((m["PricesPerCity"]/tt.shift(m["PricesPerCity"], h["Dates"], offset=120)) - 1 )* 100

m["20YRet"] = ((m["PricesPerCity"]/tt.shift(m["PricesPerCity"], h["Dates"], offset=240)) - 1 )* 100`
